### PR TITLE
Generate JSON schema definitions for web clients

### PR DESCRIPTION
**Pull request by silverjam**
To aide in building various web clients, generate JSON schema definitions to parse the output of sbp2json.  This allows web clients (like the drive demo UI) to dispatch on the `msg_type` field in a JSON dictionary and build a fully native object using the JSON fields that are present.

For example, in JavaScript, something like this would be possible:
```ts
import { Convert, MsgVelNED} from "./SbpModule.ts";

const json = getJsonBlob();

if (json['msg_type'] === 529) {
    const msgVelNED = Convert.toMsgVelNED(json);
    // ... do stuff with msgVelNED
}
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -108,7 +117,10 @@ verify-prereq-rust: ;
 
 verify-prereq-protobuf: ;
 
-verify-prereq-jsonsch: ;
+verify-prereq-jsonschema: ;
+
+verify-prereq-quicktype:
+	@command -v quicktype 1>/dev/null 2>/dev/null || { echo >&2 -e "I require \`quicktype\` but it's not installed. Aborting.\n\nHave you installed quicktype? See the generator README (Installing instructions) at \`generator/README.md\` for setup instructions.\n"; exit 1; }
```
[the remark this answers is not in the document]
Comment (denniszollo) [reply Makefile]: Can you also add the line in one of the README's for quicktype?

**Comment by denniszollo** [comment]
Can you add your note on the PR about the schema and how to use it to the javascript readme here: https://github.com/swift-nav/libsbp/blob/master/javascript/README.md too ?

File: Makefile [hunk 2]
```diff
@@ -55,6 +55,7 @@ help:
 	@echo "  java      to make Java bindings"
 	@echo "  rust      to make Rust bindings"
 	@echo "  protobuf  to make Protocol Buffer bindings"
+	@echo "  jsonsch   to make JSON Schema definitions"
```
[the remark this answers is not in the document]
Comment (silverjam) [reply Makefile]: Renamed to `jsonschema`

File: Makefile [hunk 3]
```diff
@@ -269,6 +285,10 @@ test-protobuf:
 	$(call announce-begin,"Running Protocol Buffer tests")
 	$(call announce-end,"Finished running Protocol Buffer tests")
 
+test-jsonsch:
+	$(call announce-begin,"Running JSON Schema tests")
```
[the remark this answers is not in the document]
Comment (silverjam) [reply Makefile]: :+1:

File: generator/sbpg/targets/jsonschema.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# Copyright (C) 2015 Swift Navigation Inc.
+# Contact: Bhaskar Mookerji <mookerji@swiftnav.com>
```
[the remark this answers is not in the document]
Comment (silverjam) [reply generator/sbpg/targets/jsonschema.py]: I've seen `dev@` being used pretty frequently, fixed

File: Makefile [hunk 5]
```diff
@@ -41,23 +41,28 @@ help:
...
+	@echo "  dist      		to distribute packages"
+	@echo "  docs      		to make HTML and pdf documentation"
+	@echo "  html      		to make all HTML language docs"
+	@echo "  pdf       		to make SBP LaTeX datasheet"
+	@echo "  python    		to make Python bindings"
+	@echo "  pythonNG  		to make Python (JIT) bindings"
+	@echo "  haskell   		to make Haskell bindings"
+	@echo "  java      		to make Java bindings"
+	@echo "  rust      		to make Rust bindings"
+	@echo "  protobuf  		to make Protocol Buffer bindings"
```
[the remark this answers is not in the document]
Comment (silverjam) [reply Makefile]: fixed

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (silverjam) [reply Makefile]: No particular version is required (yet).  Added some notes to README and HOWTO.

**Comment by silverjam** [comment]
merging, failure was an (errant) performance regression in python

**Pull request merged**